### ISSUE TITLE
kcllang/kcl:latest docker image has problems processing variables.

### ISSUE THREAD

**Issue by cr7258** [kcl-lang/kcl#1108]
## Bug Report

I'm practicing the tutorial according to this guide: [Implement GitOps with KCL and ArgoCD](https://kcl-lang.io/docs/next/user_docs/guides/gitops/gitops-quick-start). However, the deployment manifest can't be applied to the cluster properly with the following error:

<img width="1631" alt="image" src="https://github.com/kcl-lang/kcl/assets/40051120/8cf792a2-176c-4fe0-a049-de97e0356d23">

After checking the YAML, I found that the container name variable was not rendered correctly. The expected output was `name: guestbook` instead of `guestbook: guestbook`.

<img width="1343" alt="image" src="https://github.com/kcl-lang/kcl/assets/40051120/eafdeb68-d27b-4b49-8916-9b1707167f49">

It seems the kcllang/kcl:latest docker image has problems processing variables or is outdated.

### 1. Minimal reproduce step (Required)

```bash
https://github.com/kcl-lang/kcl-lang.io.git
cd kcl-lang.io/examples/gitops/config
```

**Run locally:**
```
kcl run
```
<img width="643" alt="image" src="https://github.com/kcl-lang/kcl/assets/40051120/76368fc8-0046-421b-81f6-fcc620d2b071">

**Using kcllang/kcl:latest docker image:**

```bash
docker run -v $(PWD):/config -w /config kcllang/kcl:latest kcl run
```
<img width="700" alt="image" src="https://github.com/kcl-lang/kcl/assets/40051120/bda2c415-e07a-4211-b49e-4e925a5177d3">

### 2. What did you expect to see? (Required)
The variable should be rendered as `name: guestbook` when use the kcllang/kcl:latest docker image.

**Comment by Peefy** [comment]
Good Catch! This is as expected, as fixing this issue in the latest version (KCL v0.8) resulted in https://github.com/kcl-lang/kcl/issues/1061 Therefore, in version 0.8 of kcl, we should appropriately fix the configuration code. PRs welcome! ❤️

```python
            template.spec.containers = [
                {
                    "name" = name # Quote the name attribute instead of name = name
```
